include/modem/pdn.h

```diff
@@ -85,6 +85,11 @@ enum pdn_auth {
  */
 typedef void (*pdn_event_handler_t)(uint8_t cid, enum pdn_event event, int reason);
 
+/**
+ * @brief Initialize PDN library.
+ */
+void pdn_init(void);
```

Comment:
Network initialization occurs at `POST_KERNEL`, before the PDN library is initialized (at `APPLICATION`. This means that `pdn_default_ctx_cb_reg` is called before `pdn_init`. This is a problem because the internal linked list (in pdn library) is reset in `pdn_init` and appended to in `pdn_default_ctx_cb_reg`. This prevents any handlers registered before pdn init from ever being invoked.

The way I see it there are two options:

Statically initialize the linked list calling `SYS_SLIST_STATIC_INIT`.
Note here is that even if we do so it might be that `on_cgev`/`on_cnec_esm` is called prior to `pdn_init` preventing handling of these AT commands due to AT monitor being paused for those commands until `pdn_init`is called.
Although I have never seen that.

Actually, the `pdn_init` can be completely removed by using static initialization. See this commit:
https://github.com/simensrostad/fw-nrfconnect-nrf/commit/28fde6e2937528821e9630211a972a26c996ea67
IMO thats the cleanest approach.

_Second option_ is the current approach where you have a `CONFIG_PDN_SYS_INIT` option and control the sequence in the connectivity layer.